### PR TITLE
MIM-15: Synka leasing db med publicerade bilplatser från xpand

### PR DESCRIPTION
**Pull request by momentiris**
Adds the possibility to sync internal parking spaces from Xpand SOAP API.

Returns an aggregated result consisting of:
- Parking spaces that we couldn't parse for some reason (missing PublishedTo being one known issue)
- Inserted Listings
- Non-inserted parking spaces

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -6,11 +6,12 @@ module.exports = {
     '<rootDir>/build/',
     //todo: excludes the db tests so that GH-action does not remove data by accident
     //todo: remove below line when test db connection exists for GH-action
-    '<rootDir>/src/services/lease-service/tests/adapters/',
+    // '<rootDir>/src/services/lease-service/tests/adapters/',
```
Comment (momentiris) [inline jest.config.js:9]: Revert this

File: src/services/lease-service/sync-internal-parking-space-listings-from-xpand.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    address: item.Address1,
+    monthlyRent: item.MonthRent,
+    objectTypeCaption: item.ObjectTypeCaption,
+    objectTypeCode: item.ObjectTypeCode,
+    rentalObjectTypeCaption: item.RentalObjectTypeCaption,
+    rentalObjectTypeCode: item.RentalObjectTypeCode,
+    publishedFrom: new Date(item.PublishedFrom),
+    publishedTo: new Date(item.PublishedTo),
+    vacantFrom: new Date(item.VacantFrom),
+    // TODO: Can we trust status should be active?
```
Comment (momentiris) [inline src/services/lease-service/sync-internal-parking-space-listings-from-xpand.ts:94]: Idk about this. Does status need to be calculated or can we trust all published parkign spaces from Xpand SOAP API should be considered active?

Comment (momentiris) [reply src/services/lease-service/sync-internal-parking-space-listings-from-xpand.ts]: Yes we can

File: jest.config.js [hunk 3]
```diff
@@ -7,10 +7,12 @@ module.exports = {
...
     //todo: remove below line when test db connection exists for GH-action
     '<rootDir>/src/services/lease-service/tests/adapters/',
+    '<rootDir>/src/services/lease-service/tests/sync-internal-parking-space-listings-from-xpand.test.ts',
   ],
   //todo: maxWorkers: 1 runs all tests in sequence so that we don't get deadlocks for db tests
   //todo: implement a more elegant solution (run db tests in sequence, all other tests in parallel)
   maxWorkers: 1,
   transformIgnorePatterns: ['node_modules/(?!(onecore-types)/)'],
   extensionsToTreatAsEsm: ['.d.ts, .ts'],
+  setupFiles: ['<rootDir>/.jest/common.ts'],
```
Comment (driatic) [inline jest.config.js:17]: 💯

File: src/services/lease-service/adapters/listing-adapter.ts [hunk 4]
```diff
@@ -45,29 +46,56 @@ function transformDbApplicant(row: DbApplicant): Applicant {
...
+        PublishedTo: listingData.publishedTo,
+        VacantFrom: listingData.vacantFrom,
+        Status: listingData.status,
+        WaitingListType: listingData.waitingListType,
+      })
+      .returning('*')
+
+    return { ok: true, data: transformFromDbListing(insertedRow[0]) }
+  } catch (err) {
+    if (err instanceof RequestError) {
```
Comment (driatic) [inline src/services/lease-service/adapters/listing-adapter.ts:76]: Great!

File: src/services/lease-service/sync-internal-parking-space-listings-from-xpand.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    parseInternalParkingSpacesToInsertableListings(internalParkingSpaces)
+
+  const withResidentialArea = await Promise.all(
+    parseInternalParkingSpacesResult.ok.map(async (v) => {
+      const result = await leaseAdapter.getResidentialAreaByRentalPropertyId(
+        v.rentalObjectCode
+      )
+
+      if (!result.ok) {
+        return Promise.reject('oops')
```
Comment (driatic) [inline src/services/lease-service/sync-internal-parking-space-listings-from-xpand.ts:88]: I guess this is just a generic error but `oops` does make me a bit confused 😄

**Review by driatic — APPROVED** [review]
Looks great to me!

**Pull request merged**